upstairs/src/lib.rs

```diff
@@ -9414,10 +9427,10 @@ struct Repair {
  * Send a message there is repair work to do.
  */
 pub struct Target {
-    ds_work_tx: mpsc::Sender<u64>,
-    ds_done_tx: mpsc::Sender<u64>,
-    ds_active_tx: watch::Sender<u64>,
-    ds_reconcile_work_tx: watch::Sender<u64>,
+    ds_work_tx: mpsc::Sender<u64>, // guest work ID?  lastcast?
```

Comment:
The comment above this struct is out of date.
`ds_work_tx`: Used to indicate that new work has arrived for a downstairs.  
The value for the u64 here is not used, and this could probably be replaced with a doorbell like mechanism.  At some point in the past we may have actually been sending the `JobId` over the channel, but that has long since passed.

`ds_done_tx`:  This is used when the upstairs has decided that we can send a response to the guest that a JobID has completed.  
As you determined, the actual JobId is not used here, as the other side of the channel will figure it out.

`ds_active_tx`: Used to indicate to each of the downstairs that the upstairs has received an activation request, and each downstairs task should take action on that request.  
The value sent is the generation number, but, again, not used.

`ds_reconcile_work_tx`: Used to send notification to downstairs tasks that there is reconciliation work to do.  The
same basic idea as `ds_work_tx`.

Reply:
Thanks, updated in 8fe2ad92e34d9c27126ffa17da5322a71456766e

At some point, I may go through and turn them all into `notify` (or `watch::Sender<()>`), but this PR touches enough stuff already.

Reply:
Yeah, feel free to create some issues and the rest of us can pick of stuff like this too.  You don't have to fix
everything by yourself :)
 But, agreed, too many unrelated changes can make future bisection difficult.